### PR TITLE
Upgrade to v2.6.0

### PR DESCRIPTION
**Pull request by metinavci**
SCLJ-47 Ability to enable/disable all library configurable from sample POS
SCLJ-53 Some pay at table configuration settings not taking effect on EFTPOS terminal

### CODE REVIEW

File: client/src/main/java/com/assemblypayments/spi/Spi.java [hunk 1]
```diff
@@ -2033,11 +2038,33 @@ public void run() {
                 DeviceService deviceService = new DeviceService();
                 DeviceAddressStatus addressResponse = deviceService.retrieveService(serialNumber, deviceApiKey, acquirerCode, inTestMode);
 
-                if (addressResponse == null) return;
+                if (addressResponse == null) {
+                    DeviceAddressStatus state = new DeviceAddressStatus();
+                    state.setAddress(null);
```
[the remark this answers is not in the document]
Comment (metinavci) [reply client/src/main/java/com/assemblypayments/spi/Spi.java]: Agreed. But, there is no setting the address for SPI. It only triggered the delegation with null DeviceAddressStatus's parameters.

[the remark this answers is not in the document]
Comment (metinavci) [reply client/src/main/java/com/assemblypayments/spi/Spi.java]: Btw, i removed the unnecessary setting the parameters to null.

File: client/src/main/java/com/assemblypayments/spi/Spi.java [hunk 2]
```diff
@@ -2033,11 +2025,38 @@ public void run() {
                 DeviceService deviceService = new DeviceService();
                 DeviceAddressStatus addressResponse = deviceService.retrieveService(serialNumber, deviceApiKey, acquirerCode, inTestMode);
 
-                if (addressResponse == null) return;
+                if (addressResponse == null) {
```
Comment (RaymondK-AP) [inline client/src/main/java/com/assemblypayments/spi/Spi.java]: Can you please change this to make the logic consistent with .NET library

Comment (metinavci) [reply client/src/main/java/com/assemblypayments/spi/Spi.java]: it's done

Comment (RaymondK-AP) [inline client/src/main/java/com/assemblypayments/spi/Spi.java]: if (addressResponse == null) || (addressResponse.getAddress() == null)

Comment (metinavci) [reply client/src/main/java/com/assemblypayments/spi/Spi.java]: it works like that. There is no crash as .Net

File: client/src/main/java/com/assemblypayments/spi/Spi.java [hunk 3]
```diff
@@ -2025,7 +2025,7 @@ public void run() {
                 DeviceService deviceService = new DeviceService();
                 DeviceAddressStatus addressResponse = deviceService.retrieveService(serialNumber, deviceApiKey, acquirerCode, inTestMode);
 
-                if (addressResponse == null) {
+                if (addressResponse == null || (addressResponse.getAddress() == null && (addressResponse.getResponseCode() != 200 && addressResponse.getResponseCode() != 404))) {
```
Comment (RaymondK-AP) [inline client/src/main/java/com/assemblypayments/spi/Spi.java]: Do you need to check addressResponse.getResponseCode() == null ?

Comment (metinavci) [reply client/src/main/java/com/assemblypayments/spi/Spi.java]: yes should be

**Pull request merged**